libsolidity/interface/CompilerStack.cpp

```diff
@@ -64,6 +65,11 @@ using namespace dev;
 using namespace langutil;
 using namespace dev::solidity;
 
+CompilerStack::~CompilerStack()
+{
+	TypeProvider::get().reset();
```

Comment:
This basically means that `TypeProvider` is "owned" by CompilerStack. Can we make that more explicit? For example in that the CompilerStack "creates" the singleton and thus ensures that a) it is not used before CompilerStack is instanciated and b) it is not "owned" by more than one CompilerStack?